### ISSUE TITLE
tIED: insertIed incorrectly orders Communication and DataTypeTemplates sections

### ISSUE THREAD

**Issue by danyill** [OpenEnergyTools/scl-lib#46]
When I use a "stub" file as the basis for testing an import of a template IED into a file which has neither an existing `Communication` or `DataTypeTemplates` section, these are mis-ordered and not schema valid in the resulting file.

We start with:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<SCL xmlns="http://www.iec.ch/61850/2003/SCL" version="2007" revision="B" release="4">
	<Header id="test_comms" />
</SCL>
```

And import an IED.

We find that the `Communication` section occurs after the `DataTypeTemplates` section.

I believe this is due to the order of operations:

https://github.com/OpenEnergyTools/scl-lib/blob/864bb51c14137a04d70199d9df3d0e83a9531b07/tIED/insertIED.ts#L296-L315

We should (I think) always follow the order:

* Communication
* IED
* DataTypeTemplates 

to avoid absent sections causing an "order of operations" issue in the way our elements are added.

**Comment by JakobVogelsang** [comment]
Uff that is a shame. Your proposal won't be enough, I feel. The problem is that we determine `reference` statically before sending the edit event. We need to better determine static references. That is a tricky one :(

**Comment by JakobVogelsang** [comment]
Fixed with release 0.5.11

https://github.com/OpenEnergyTools/scl-lib/commit/a6577af4b11c242a8ceda1e89e0f5ae4363556a1

**Comment by danyill** [comment]
The solution looks good, thanks :+1:
